store/rootmulti/store.go

```diff
@@ -192,7 +192,20 @@ func (rs *Store) loadVersion(ver int64, upgrades *types.StoreUpgrades) error {
 	// load each Store (note this doesn't panic on unmounted keys now)
 	var newStores = make(map[types.StoreKey]types.CommitKVStore)
 
-	for key, storeParams := range rs.storesParams {
+	storesKeys := make([]types.StoreKey, 0, len(rs.storesParams))
+
+	for key := range rs.storesParams {
+		storesKeys = append(storesKeys, key)
+	}
+	if upgrades != nil {
+		// deterministic iteration order for upgrades
+		sort.Slice(storesKeys, func(i, j int) bool {
+			return strings.Compare(storesKeys[i].Name(), storesKeys[j].Name()) == -1
```

Comment:
We can simply use < and thus do:

return storeKeys[i].Name() < storeKeys[j].Name()